validator/stateless_block_validator.go

```diff
@@ -307,11 +307,17 @@ func SetMachinePreimageResolver(ctx context.Context, mach *ArbitratorMachine, pr
 			if err != nil {
 				log.Error("Failed to deserialize DAS message", "err", err)
 			} else {
-				dasPreimage, err := das.Retrieve(ctx, seqMsg[40:])
+				keysetPreimage, err := das.KeysetFromHash(ctx, cert.KeysetHash[:])
```

Comment:
Noting that `KeysetFromHash` doesn't guarantee return value that matches the keyset hash since it fetches it from a remote server. Also I think not being able to get the keyset for a given hash needs to be a hard error since it that case the chain will be stuck since we can't decide if a batch is validly signed